plugins/plugin-testing-java/plugin-testing-junit/che-plugin-testing-junit-server/src/main/java/org/eclipse/che/plugin/testing/junit/server/listener/OutputTestListener.java

```diff
@@ -0,0 +1,105 @@
+/*******************************************************************************
+ * Copyright (c) 2017 RedHat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   RedHat, Inc. - initial commit
+ *******************************************************************************/
+package org.eclipse.che.plugin.testing.junit.server.listener;
+
+import static org.eclipse.che.api.testing.shared.Constants.TESTING_OUTPUT_CHANNEL_NAME;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+
+import org.eclipse.che.api.core.util.WebsocketMessageConsumer;
+import org.eclipse.che.api.testing.server.handler.TestingOutputImpl;
+import org.eclipse.che.api.testing.shared.TestingOutput;
+import org.eclipse.che.dto.server.DtoFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Listener for the testing services to report their progress to the Che output view.
+ *
+ * @author David Festal
+ */
+public class OutputTestListener extends AbstractTestListener implements AutoCloseable {
+    private static final Logger LOG = LoggerFactory.getLogger(AbstractTestListener.class);
```

Comment:
That will be, surely, but is it required *now*, or could I push the tests in a next pull request ?